audio/stereo_panner.rs

```diff
@@ -0,0 +1,139 @@
+        Self {
+            pan: options.pan,
+        }
+    }
+
+    pub fn update_parameters(&mut self, info: &BlockInfo, tick: Tick) -> bool {
+        self.pan.update(info, tick)
+    }
+}
+
+impl AudioNodeEngine for StereoPannerNode {
+    fn node_type(&self) -> AudioNodeType {
+        AudioNodeType::StereoPannerNode
+    }
+
+    fn process(&mut self, mut inputs: Chunk, info: &BlockInfo) -> Chunk {
+        debug_assert!(inputs.len() == 1);
+
+        return inputs;
+
+        {
+            let block = &mut inputs.blocks[0];
+
+            block.explicit_repeat();
+
+            let mono = if block.chan_count() == 1 {
+                block.resize_silence(2);
+                true
+            } else {
+                debug_assert!(block.chan_count() == 2);
+                false
+            };
+
+                let (l, r) = block.data_mut().split_at_mut(FRAMES_PER_BLOCK.0 as usize);
+                for frame in 0..FRAMES_PER_BLOCK.0 {
+                    let frame = Tick(frame);
+                    self.update_parameters(info, frame);
+                    let mut pan = self.pan;
+
+                    // https://webaudio.github.io/web-audio-api/#stereopanner-algorithm
+
+                    // clamp pan to [-1, 1]
+                    pan = max(-1, pan);
+                    pan = min(1, pan);
+
+                    let x = if mono {
+                        x = (pan + 1) / 2;
+                    } else if pan <= 0. {
+                        x = pan + 1;
+                    } else {
+                        x = pan;
+                    };
+                    let x = x * PI / 2.;
+
+                    let mut gain_l = x.cos();
+                    let mut gain_r = x.sin();
+                    // 9. * PI / 2 is often slightly negative, clamp
+                    if gain_l <= 0. {
+                        gain_l = 0.
+                    }
+                    if gain_r <= 0. {
+                        gain_r = 0.;
+                    }
+
+                    let index = frame.0 as usize;
+                    if mono {
+                        let input = l[index];
+                        l[index] = input * gain_l;
+                        r[index] = input * gain_r;
+                    } else if pan <= 0. {
+                        l[index] = l[index] + r[index] * gain_l;
+                        r[index] = r[index] * gain_r;
+                    } else {
+                        r[index] = r[index] + l[index] * gain_r;
+                        l[index] = l[index] * gain_l;
+                    }
+                }
+
+        }
+
+        inputs
+    }
+
+    fn input_count(&self) -> u32 {
+        1
+    }
+
+    fn get_param(&mut self, id: ParamType) -> &mut Param {
+        match id {
+            ParamType::Position(Param::pan) => &mut self.pan,
+            _ => panic!("Unknown param {:?} for PannerNode", id),
+        }
+    }
+
+    fn message_specific(&mut self, message: AudioNodeMessage, _sample_rate: f32) {
```

Comment:
This function isn't necessary, the `get_param` deals with all of this. You only need this if you have non-Param "parameters"